target/esp32/ll_cam.c

```diff
@@ -382,13 +382,13 @@ uint8_t ll_cam_get_dma_align(cam_obj_t *cam)
 }
 
 static bool ll_cam_calc_rgb_dma(cam_obj_t *cam){
-    size_t dma_half_buffer_max = 16 * 1024 / cam->dma_bytes_per_item;
+    size_t dma_half_buffer_max = CONFIG_CAMERA_DMA_BUFFER_SIZE_MAX / 2 / cam->dma_bytes_per_item;
```

Comment:
with the default config, this size drops from 16k to 4k. Why so low?

Reply:
Because in my tests, I found that sometimes failed by couldn't allocate 32000bytes(total buffer size) internal RAM. Moreover, the size of this buffer does not affect the transmission speed at all. Too large buffer will occupy too much memory

Reply:
maybe it should detect if it failed to allocate and lower the size instead? lowering this 4 times has other consequences as well, like getting interrupts and events 4 times more often. each comes with context switches and whatnot. Let's default to the larger size and lower it if necessary (config value could be noted in the error log)

Reply:
The code has been updated as suggested above